Manual TxBuilder

mankyKitty opened the pull request.

Commit:
WIP: Broken, deconstructing txbuilder to definedkeyset

Commit:
Added widget, valid txbuilder updates defined keyset widget.

Commit:
Working exploded txbuilder widget integrated to send dialog.

Commit:
Better living through semigroups

Commit:
Confirm the external public keys in the txbuilder are valid.

Pact does not validate public keys on JSON decode

Commit:
Remove unused language pragma

Commit:
Refactor manual txbuilder

Commit:
Add reset to txbuilder field when manual exploded fields are altered

Add popover error feedback to exploded txbuilder fields.

alexfmpe commented on frontend/src/Frontend/UI/Widgets.hs:
```diff
@@ -878,29 +883,39 @@ uiGasPriceInputField eReset conf = dimensionalInputFeedbackWrapper (Just "KDA")
   -> CssClass
-  -> m (Dynamic t (Maybe ChainId))
+  -> m ( Dropdown t (Maybe ChainId)
+       , Dynamic t (Maybe ChainId)
```
Unlike when we return `(Event t, Dynamic t)`, these are redundant, but callers don't know that. I'd just return `Dropdown t`.

Commit:
Only return `Dropdown` from ChainId select

Commit:
Track updates from individual fields from uiDefineKeyset

Track updates from the individual fields so that the txbuilder input can be reset on field changes.
